lib/generate-index.js

```diff
@@ -12,22 +12,22 @@ const { decode } = require('html-entities')
  * @memberof generate-index
  *
  * @param {Object} playbook - The configuration object for Antora.
- * @param {Array<File>} pages - The publishable pages to map.
- * @param {Object} contentCatalog - the Antora content catalog (allows access to page metadata).
- * @param {Object} env - command line environment variables.
+ * @param {Object} contentCatalog - The Antora content catalog, with pages and metadata.
+ * @param {Array<String>} config.languages - List of index languages
+ * @param {Boolean} config.languages - If true, only index the latest version of any given page.
  * @returns {Object} A JSON object with a Lunr index and a documents store.
  */
-function generateIndex (playbook, pages, contentCatalog, env) {
+function generateIndex (playbook, contentCatalog, { languages, indexLatestOnly }) {
   let siteUrl = playbook.site.url
   if (!siteUrl) {
     // Uses relative links when site URL is not set
     siteUrl = ''
   }
   if (siteUrl.charAt(siteUrl.length - 1) === '/') siteUrl = siteUrl.substr(0, siteUrl.length - 1)
-  if (!pages.length) return {}
+  if (!contentCatalog.getPages().length) return {}
```

Comment:
It's possible for this method to return pages which are not publishable. This list needs to be filtered as follows:

```js
contentCatalog.getPages((page) => page.out)
```

I also think this method should only be called once. So grab the documents first, then check if the collection is empty.

Reply:
Thanks for both of those, @mojavelinux. Done!